charts/cluster-api-cluster-openstack/values.yaml

```diff
@@ -73,19 +73,22 @@ openstack:
 # Modifications to this object will trigger a control plane upgrade.
 controlPlane:
   # Version of Kubernetes, should match that installed on the base images.
-  version: v1.25.2
+  version: v1.30.2
 
   # Number of control plane machines.
   replicas: 3
 
+  # Skips the kube proxy init phase in kubeadm.
```

Comment:
What does that actually mean, and what are the repercussions for someone setting it?  Sorry to be a pain, just if I'm confused, other users will be and that will lead to questions and time wasting.

Reply:
ok so if you take a look here:
https://kubernetes.io/docs/reference/setup-tools/kubeadm/kubeadm-init/

This flag essentially allows the kube-proxy (which is considered an addon) to be skipped during initialisation. Setting this means that kube-proxy won't be deployed which in turn means it won't be there to handle iptables - (which is slow and clunky and scales terribly).

To enable Cilium to take control of the loadbalancing at the eBPF level, this needs to be disabled
https://cilium.io/use-cases/kube-proxy/

So, if a user disables this, then they'd have to have a mechinism in place, such as Cilium, to handle this traffic. 

I've deployed this with the flag enabled and the cluster will sit waiting for the CNI as it usually does anyway so no impact to deployment in the flow of things. If the user doesn't deploy something that can't handle the load balancing that kube-proxy supplies, such as Cilium, then nothing will route correctly. 


It's expected a user should know why they want this when enabling this - hence it being off by default.

Reply:
Another option would be to map a `kubeadmSkipPhases` list to the [skipPhases here ](https://github.com/drew-viles/helm-cluster-api/blob/update-docs/charts/cluster-api-cluster-openstack/templates/control-plane.yaml#L31-L32)allowing a user to supply these themselves rather than limiting it to KubeProxy?

Reply:
In fact, that is a better approach on the whole - I've a better approach with a better comment incoming - hold please!

Reply:
See what you think of that change instead. Better comment explaining what it is and leave it to the user to add the phases they wish to skip - just means if they don't want coredns, for example, they can skip that too and prevents subsequest "requests for features".